app/src/main/java/org/mozilla/fenix/FenixApplication.kt

```diff
@@ -80,7 +80,7 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
             return
         }
 
-        if (Config.channel.isFenix) {
+        if (Config.channel.isFenix && BuildConfig.TELEMETRY) {
```

Comment:
If I reasoned correctly, this should leave telemetry enabled for Nightly, as it `isFenix` but `!isDebug`, unless `TELEMETRY` is disabled. That also means that `TELEMETRY` is disabled when debugging. I believe this is the correct behavior that we are looking for.

Reply:
Yes -- and it leaves a single build flag for forks that which to disable telemetry entirely.